### ISSUE TITLE
[a11y]: Select usage and other guidance updates to support accessibility

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon-website#3361]
This issue captures the high-level considerations from a review of the Usage, Style, storybook and figma documentation on the Select component. More details can be discovered by examing in the comments in the pdf exports of the [Usage](https://ibm.box.com/s/6zkqgvu29ctw52mhfpq7op1l2x413qxj) and [Style](https://ibm.box.com/s/rs7cqkd5bcdi0dh1d8ptru4acxn1gzdd) pages.

It is complementary to the https://github.com/carbon-design-system/carbon/issues/13008 issue covering actual implementations with the Select component in the React storybook.

1. Various suggested rewrites for clarity and accuracy, commented on in the Usage pdf (linked to above)
2. Inconsistencies between images and captions
3. Some of the decisions on illustrations between this and the [prior version](https://ibm.box.com/s/5zetn7rglz1tmlzzalhgqy2qcq5lmrxy) of the Usage page are not improvements.
4. Calling the select list a "menu" in the anatomy section is ill-advised
5. Material in the Content section of the Usage has some challenges.
6. "Selected" is missing as a state in the Behaviors section
7. There are several inaccuracies in the Style content.

**Comment by sstrubberg** [comment]
closing in favor of tracking via https://github.com/carbon-design-system/carbon/issues/13521
